jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3029,6 +3015,72 @@ public MethodHandle run() {
 	return valueMethod;
 }
 
+private MetadataCache getMetadataCache() {
+	if (metadataCache == null) {
+		metadataCacheOffset = getFieldOffset("metadataCache"); //$NON-NLS-1$
+		writeFieldValue(metadataCacheOffset, new MetadataCache());
+	}
+	return metadataCache;
+}
+
+private String cacheSimpleName(String simpleName) {
+	MetadataCache cache = getMetadataCache();
+
+	if (cache.cachedSimpleName == null || cache.cachedSimpleName.get() == null) {
+		MetadataCache.cachedSimpleNameOffset = getFieldOffset(
+				MetadataCache.class, "cachedSimpleName", MetadataCache.cachedSimpleNameOffset); //$NON-NLS-1$
```

Comment:
```suggestion
		if (-1 == MetadataCache.cachedSimpleNameOffset) {
			/* Only lookup the offset if we haven't found it already */
			MetadataCache.cachedSimpleNameOffset = getFieldOffset(
				MetadataCache.class, "cachedSimpleName", MetadataCache.cachedSimpleNameOffset); //$NON-NLS-1$
		}
```

Reply:
As discussed, `getFieldOffset` has this check already, so I will leave it as is.